e2e/suite.go

```diff
@@ -73,12 +74,34 @@ func Run(t *testing.T) {
 		return filepath.Join(buildcfg.SYSCONFDIR, "singularity", fn)
 	}
 
+	// Make temp dir for tests
+	name, err := ioutil.TempDir("", "stest.")
+	if err != nil {
+		log.Fatalf("failed to create temporary directory: %v", err)
+	}
+	defer e2e.Privileged(func(t *testing.T) {
+		if t.Failed() {
+			t.Logf("Test failed, not removing %s", name)
+			return
+		}
+
+		os.RemoveAll(name)
+	})(t)
+
+	if err := os.Chmod(name, 0755); err != nil {
+		log.Fatalf("failed to chmod temporary directory: %v", err)
+	}
+	testenv.TestDir = name
+
 	// e2e tests need to run in a somehow agnostic environment, so we
 	// don't use environment of user executing tests in order to not
 	// wrongly interfering with cache stuff, sylabs library tokens,
 	// PGP keys
 	e2e.SetupHomeDirectories(t)
 
+	// generate singularity.conf with default values
+	e2e.SetupDefaultConfig(t, filepath.Join(testenv.TestDir, "singularity.conf"))
```

Comment:
How does Singularity use this config file in TestDir at all? If we are calling an installed `singularity` executable then isn't it using the one in `sysconfdir`?

Can you explain why it's necessary to do this SetupDefaultConfig into TestDir?

Reply:
The idea is to not let the current `singularity.conf` influence tests and use the default values, that's why a config is generated as if it was the first installation of singularity and the file is bind mount on top of the current : https://github.com/sylabs/singularity/pull/4760/files#diff-c95326473ca3935b0c44e93cde3b7059R35

This way current configuration set by the dev won't conflicts with e2e tests, actually if you put `bind path = /my/nfs/share` it's bound into container during e2e run because install procedure doesn't reset configuration file but instead update it if necessary (like new directive) and reuse current value to generate the new one.